vms/event/tests/test_model.py

```diff
@@ -0,0 +1,99 @@
+# third party
+
+# Django
+from django.db.models import Q
+from django.test.testcases import TestCase
+
+# local Django
+from event.models import Event
+from shift.utils import create_event_with_details
+
+
+class EventModelTests(TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    @staticmethod
+    def create_event():
+        event = ['event-name', '2050-05-24', '2050-05-28']
+        created_event = create_event_with_details(event)
+        return created_event
+
+    def test_valid_model_create(self):
+        event = ['event-name', '2050-05-24', '2050-05-28']
+        create_event_with_details(event)
+
+        # Check database for instance creation
+        self.assertNotEqual(len(Event.objects.all()), 0)
+
+        event_in_db = Event.objects.get(Q(name='event-name'))
+        # Verify correctness
+        self.assertEqual(event_in_db.name, event[0])
+        self.assertEqual(str(event_in_db.start_date), event[1])
+        self.assertEqual(str(event_in_db.end_date), event[2])
+
+    # Checks are missing from model class
+    # `clean` functions need to be implemented.
+    # def test_invalid_model_create(self):
+    #     event = ['event-name', '2050-05-24', '2050-05-21']
+    #     create_event_with_details(event)
+    #
+    #     # Check database for instance creation
+    #     self.assertNotEqual(len(Event.objects.all()), 0)
+    #
+    #     event_in_db = Event.objects.get(Q(name='event-name'))
+    #     # Verify correctness
+
+    def test_model_edit_with_valid_values(self):
+        created_event = EventModelTests.create_event()
+
+        # Check db for instance creation
+        self.assertEqual(len(Event.objects.all()), 1)
+
+        event_in_db = Event.objects.get(Q(name='event-name'))
+        # Check if correct event retrieved
+        self.assertEqual(event_in_db.name, created_event.name)
+
+        # Edit job
+        event_in_db.name = 'new-event-name'
+        event_in_db.save()
+
+        event_in_db = Event.objects.get(Q(name='new-event-name'))
+        # Check if save is success
+        self.assertEqual(event_in_db.name, 'new-event-name')
+        self.assertEqual(len(Event.objects.all()), 1)
+
+    # Clear function needs to be implemented for this.
```

Comment:
Same here.

Reply:
Workaround found:
model validation checks aren't performed until a `full_clean` is called explicitly from code/command-line, Forms perform `full_clean` on submit thats why the error messages are shown. Thus `model.save()` will save the instance unless there not-null constraint is broken.